### PR TITLE
improve: Add improvements and fixes found when fixing relayer-v2 unit tests

### PR DESCRIPTION
**Pull request by nicholaspai**

### CODE REVIEW

File: src/clients/UBAClient/UBAClientUtilities.ts [hunk 1]
```diff
@@ -254,7 +254,7 @@ export function getMostRecentBundleBlockRanges(
       // spoke block searched.
       if (bundleData.length === 0) {
         const ubaActivationBundleStartBlock = getUbaActivationBundleStartBlocks(hubPoolClient, [chainId])[0];
-        bundleData.push({
+        bundleData.unshift({
```
Comment (james-a-morris) [inline src/clients/UBAClient/UBAClientUtilities.ts:257]: Could we build it in order with push then reverse the list?

File: src/clients/UBAClient/UBAClientUtilities.ts [hunk 2]
```diff
@@ -269,7 +269,13 @@ export function getMostRecentBundleBlockRanges(
       latestExecutedRootBundle
     );
     // Make sure our spoke pool clients have the block ranges we need to look up data in this bundle range:
-    if (blockRangesAreInvalidForSpokeClients(spokePoolClients, rootBundleBlockRanges)) {
+    if (
+      blockRangesAreInvalidForSpokeClients(
+        spokePoolClients,
+        rootBundleBlockRanges,
+        hubPoolClient.configStoreClient.enabledChainIds
```
Comment (james-a-morris) [inline src/clients/UBAClient/UBAClientUtilities.ts:276]: Good refactor. Nice reuse of logic

File: src/clients/UBAClient/UBAClientUtilities.ts [hunk 3]
```diff
@@ -279,14 +285,22 @@ export function getMostRecentBundleBlockRanges(
 
     // If UBA is not enabled for this bundle, exit early since no subsequent bundles will be enabled
     // for the UBA, as the UBA was a non-reversible change.
-    const hubPoolStartBlock = getBlockRangeForChain(rootBundleBlockRanges, hubPoolClient.chainId)[0];
+    const hubPoolStartBlock = getBlockRangeForChain(
+      rootBundleBlockRanges,
+      hubPoolClient.chainId,
+      hubPoolClient.configStoreClient.enabledChainIds
+    )[0];
     if (!isUbaBlock(hubPoolStartBlock, hubPoolClient.configStoreClient)) {
```
Comment (james-a-morris) [inline src/clients/UBAClient/UBAClientUtilities.ts:293]: Does this account for if hubPoolStartBlock is undefined?

**Review by james-a-morris — COMMENTED** [review]
Left some comments

File: src/clients/UBAClient/UBAClientUtilities.ts (hunk 3 above)
Comment (nicholaspai) [reply src/clients/UBAClient/UBAClientUtilities.ts:293]: It should never be `undefined` here and thats enforced by the type

**Review by james-a-morris — APPROVED** [review]

**Pull request merged**